src/CommunityToolkit.Maui/Views/RatingView/RatingView.Shared.cs

```diff
@@ -0,0 +1,380 @@
+	public static readonly BindableProperty ShouldAllowRatingProperty = BindableProperty.Create(nameof(ShouldAllowRating), typeof(bool), typeof(RatingView), defaultValue: false, propertyChanged: OnBindablePropertyChanged);
+
+	///<summary></summary>
+	public static readonly BindableProperty CommandProperty = BindableProperty.Create(nameof(Command), typeof(ICommand), typeof(RatingView), defaultValue: null, propertyChanged: OnBindablePropertyChanged);
+
+	///<summary></summary>
+	public static readonly BindableProperty CommandParameterProperty = BindableProperty.Create(nameof(CommandParameter), typeof(object), typeof(RatingView), defaultValue: null, propertyChanged: OnBindablePropertyChanged);
+
+	///<summary></summary>
+	public readonly BindableProperty ShapeProperty = BindableProperty.Create(nameof(Shape), typeof(RatingShape), typeof(RatingView), propertyChanged: OnShapePropertyChanged);
+
+	#endregion
+
+	#region Public Properties
+
+	///<summary></summary>
+	public double CurrentRating
+	{
+		get => (double)GetValue(CurrentRatingProperty);
+		set => SetValue(CurrentRatingProperty, value);
+	}
+
+	///<summary></summary>
+	public int MaximumRating
+	{
+		get => (int)GetValue(MaximumRatingProperty);
+		set => SetValue(MaximumRatingProperty, value);
+	}
+
+	///<summary></summary>
+	public double Size
+	{
+		get => (double)GetValue(SizeProperty);
+		set => SetValue(SizeProperty, value);
+	}
+
+	///<summary></summary>
+	public Color FilledBackgroundColor
+	{
+		get => (Color)GetValue(FilledBackgroundColorProperty);
+		set => SetValue(FilledBackgroundColorProperty, value);
+	}
+
+	///<summary></summary>
+	public Color EmptyBackgroundColor
+	{
+		get => (Color)GetValue(EmptyBackgroundColorProperty);
+		set => SetValue(EmptyBackgroundColorProperty, value);
+	}
+
+	///<summary></summary>
+	public Color StrokeColor
+	{
+		get => (Color)GetValue(StrokeColorProperty);
+		set => SetValue(StrokeColorProperty, value);
+	}
+
+	///<summary></summary>
+	public new double StrokeThickness
+	{
+		get => base.StrokeThickness;
+		set => SetValue(StrokeThicknessProperty, value);
+	}
+
+	///<summary></summary>
+	public double Spacing
+	{
+		get => (double)GetValue(SpacingProperty);
+		set => SetValue(SpacingProperty, value);
+	}
+
+	///<summary></summary>
+	public bool ShouldAllowRating
+	{
+		get => (bool)GetValue(ShouldAllowRatingProperty);
+		set => SetValue(ShouldAllowRatingProperty, value);
+	}
+
+	///<summary></summary>
+	public ICommand? Command
+	{
+		get => (ICommand)GetValue(CommandProperty);
+		set => SetValue(CommandProperty, value);
+	}
+
+	///<summary></summary>
+	public object? CommandParameter
+	{
+		get => GetValue(CommandParameterProperty);
+		set => SetValue(CommandProperty, value);
+	}
+
+	///<summary></summary>
+	public RatingShape Shape
+	{
+		get => (RatingShape)GetValue(ShapeProperty);
+		set => SetValue(ShapeProperty, value);
+	}
+
+	///<summary></summary>
+	public new View? Content { get => base.Content; internal set => base.Content = value; }
```

Comment:
why can't we use the `base.Content`? Doing this isn't very safe, if someone cast this to `Border` (base type) we will use the base.Content instead of this one, so this isn't safe

Reply:
Here's [a code](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEUCuA7AHwAEAmABgFgAoUgRmqLIAIjakXaA6AYQjwGcIAGxgBuag1oBOABQAiQDwbgEH25ASnFVqAMQgQmAMyYBeJnhgB3Jjogz1EqgHUoASwwwAMs7Mz9PPm7wMO00qADcAQygmMHD+DGMmGQAhWJgAFVV9DWonVw8vGBkYuL9AmEDg6mdDHyZnfiYU/nSmYFSM6gBvaiZeply3T2825rTSgKCNPtb28fL4kzkAOQsmXjLAuSm+gfzh2fWJ9WnqAF8evvsrmhIrXSYQRvaui5YAZlNV1mZD+aZOgDmMAwoiYAAcXBE3ExmiDTgk5FooBAALZrAAWziEABMtmdrqQnqMXlRpkQPt81v4/oDgaDYaJ4YskaiZs08VRziFqEA=) that shows I meant.

We can document that the `base.Content` will not affect the control itself... But not sure how that will behave if devs go too fancy.

Or maybe `Border` isn't the right control to use for RatingView

Reply:
@Eel2000 Please take a look how .NET MAUI team implemented IndicatorView. Rating View should be very similar

Reply:
Thank for the hint.

Reply:
Please @VladislavAntonyuk , I noticed that the IndicatorView uses a TemplatedView which uses compatibility lib, as mentioned here https://github.com/CommunityToolkit/Maui/issues/1607#issuecomment-1947314984. Should I still use it? Because that's what the original ratingView's code uses

Reply:
I mean it can just be a StackLayout with BindableLayouts: https://github.com/VladislavAntonyuk/maui/blob/indicatorview-itemtemplate-binding/src/Controls/src/Core/IndicatorView/IndicatorStackLayout.cs

Reply:
@VladislavAntonyuk , `StackLayout` is under compatibility layers as well

Reply:
*HorizontalStackLayout

Reply:
> @VladislavAntonyuk , `StackLayout` is under compatibility layers as well

if tha so, then i can still use the templatedView?? its simplify the implementation itself.